client/web/src/site-admin/RepositoryNode.tsx

```diff
@@ -35,24 +57,61 @@ const RepositoryStatusBadge: React.FunctionComponent<{ status: string }> = ({ st
     </Badge>
 )
 
-const parseRepositoryStatus = (repository: SiteAdminRepositoryFields): string => {
+const parseRepositoryStatus = (repo: SiteAdminRepositoryFields): string => {
     let status = 'queued'
-    if (repository.mirrorInfo.cloned && !repository.mirrorInfo.lastError) {
+    if (repo.mirrorInfo.cloned && !repo.mirrorInfo.lastError) {
         status = 'cloned'
-    } else if (repository.mirrorInfo.cloneInProgress) {
+    } else if (repo.mirrorInfo.cloneInProgress) {
         status = 'cloning'
-    } else if (repository.mirrorInfo.lastError) {
+    } else if (repo.mirrorInfo.lastError) {
         status = 'failed'
     }
     return status
 }
 
+const repoClonedAndHealthy = (repo: SiteAdminRepositoryFields): boolean =>
+    repo.mirrorInfo.cloned && !repo.mirrorInfo.lastError && !repo.mirrorInfo.cloneInProgress
+
 interface RepositoryNodeProps {
     node: SiteAdminRepositoryFields
 }
 
+const updateNodeFromData = (node: SiteAdminRepositoryFields, data: SettingsAreaRepositoryResult | undefined) => {
+    if (data?.repository && data.repository?.mirrorInfo) {
+        node.mirrorInfo.lastError = 'bad errror 123'
```

Comment:
I like your option, but we can use a more boring one :)

```suggestion
        node.mirrorInfo.lastError = data.repository.mirrorInfo.lastError
```

Reply:
Whoooooooooops

Reply:
It’s what I get for late night commits